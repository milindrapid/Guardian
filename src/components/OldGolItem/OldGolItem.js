import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import '../../fontawesome';

const OldGolItemContainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 0.5rem;
    p {
        margin-left: 1.75rem;
        margin-top: 0.75rem;
        font-style: normal;
        font-weight: normal;
        font-size: 1rem;
        line-height: 1.375rem;
    }
`;

const OldGolItemHeader = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    a {
        font-style: normal;
        font-weight: bold;
        font-size: 1rem;
        line-height: 1.375rem;
        text-decoration: underline;
        color: #0D3F5E;
        cursor: pointer;
    }
`;

const OldGolItemIcon = styled.span`
    display: block;
    width: 1.25rem;
    height: 1.25rem;
    text-align: center;
    font-style: normal;
    font-weight: 900;
    font-size: 1rem;
    line-height: 1.125rem;
    margin-right: 0.5rem;
`;

const OldGolItem = (props) => {
    const { className } = props;
    const { 
        itemName,
        itemDescription,
        iconName,
        redirectUrl
     } = props.oldGolItem;

     return (
         <OldGolItemContainer className={className}>
             <OldGolItemHeader>
                <OldGolItemIcon>
                    <FontAwesomeIcon icon={iconName} />
                </OldGolItemIcon>
                <a href={redirectUrl}>{itemName}</a>
             </OldGolItemHeader>
             <p>{itemDescription}</p>
         </OldGolItemContainer>
     )
};

OldGolItem.propTypes = {
    className: PropTypes.string,
    oldGolItem: PropTypes.shape({
        itemName: PropTypes.string.isRequired,
        itemDescription: PropTypes.string.isRequired,
        iconName: PropTypes.string.isRequired,
        redirectUrl: PropTypes.string
    })
};

OldGolItem.defaultProps = {
    oldGolItem: {
        redirectUrl: "/",
    }
};

export default OldGolItem;