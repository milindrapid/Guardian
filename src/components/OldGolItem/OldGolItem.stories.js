import React from "react";
import OldGolItem from "./OldGolItem";

export default {
    title: 'Old GOL Item',
    component: 'OldGolItem'
};

const bookOfBusinessProps = {
    itemId: 1,
    itemName: 'Book of Business',
    itemDescription: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
    iconName: 'chart-line'
};

const eServicesProps = {
    itemId: 2,
    itemName: 'eServices',
    itemDescription: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
    iconName: 'clipboard-list'
};

const trackNewBusinessProps = {
    itemId: 3,
    itemName: 'Track New Business',
    itemDescription: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
    iconName: 'briefcase'
};

const clientManagerProps = {
    itemId: 4,
    itemName: 'Client Manager',
    itemDescription: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
    iconName: 'user'
};

export const bookOfBusinessCard = () => <OldGolItem oldGolItem={bookOfBusinessProps} />
export const eServicesCard = () => <OldGolItem oldGolItem={eServicesProps} />
export const trackNewBusinessCard = () => <OldGolItem oldGolItem={trackNewBusinessProps} />
export const clientManagerCard = () => <OldGolItem oldGolItem={clientManagerProps} />


bookOfBusinessCard.story = {
    name: 'Book of Business'
};

eServicesCard.story = {
    name: 'eServices'
};

trackNewBusinessCard.story = {
    name: 'Track New Business'
};

clientManagerCard.story = {
    name: 'Client Manager'
};