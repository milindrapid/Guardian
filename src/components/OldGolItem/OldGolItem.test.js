import React from "react";
import OldGolItem from "./OldGolItem";
import { shallow } from "enzyme";

describe("EventCard component", () => {
    const props = {
        itemId: 1,
        itemName: 'Book of Business',
        itemDescription: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
        iconName: 'chart-line'
    };
    
  const wrapperComponent = shallow(<OldGolItem oldGolItem={props} />);

  it("Old GOL item rendered correctly", () => {
    expect(wrapperComponent).toMatchSnapshot();
  });

  it("should render children components correclty", () => {
    expect(wrapperComponent.find('FontAwesomeIcon')).toHaveLength(1);
    expect(wrapperComponent.find('a')).toHaveLength(1);
    expect(wrapperComponent.find('p')).toHaveLength(1);
  });

//   it("should render children components correclty", () => {
//     props.isLive = true;
//     wrapperComponent.setProps({eventDetails: props});
//     expect(wrapperComponent.find(LiveIcon)).toHaveLength(1);
//     expect(wrapperComponent.find("label").text()).toMatch("LIVE");
//   });
});
