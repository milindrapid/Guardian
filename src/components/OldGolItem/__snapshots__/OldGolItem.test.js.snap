// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard component Old GOL item rendered correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="chart-line"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.span>
    <a>
      Book of Business
    </a>
  </styled.div>
  <p>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </p>
</styled.div>
`;
